### PR TITLE
Fixing errors flagged by PhpStan [For Release 3_4]

### PR DESCRIPTION
**Pull request by drillinP**
* Getting PhpStan from @alagroy-42  [commit](https://github.com/3liz/lizmap-web-client/pull/2029/commits/e71a15e9f00758e304ad72eb1fd7a40539432570)
* Fixing errors flagged by PhpStan

### CODE REVIEW

File: lizmap/modules/action/classes/actionDatasource.class.php [hunk 1]
```diff
@@ -41,11 +41,11 @@ public function __construct($repository, $project, $layerId)
         jClasses::inc('action~actionConfig');
         $dv = new actionConfig($repository, $project);
         if (!$dv->getStatus()) {
-            return $this->error($dv->getErrors());
+            return $dv->getErrors();
```
Comment (drillinP) [inline lizmap/modules/action/classes/actionDatasource.class.php:44]: @rldhont **Phpstan** say  the `error` method doesn't exists, so i think we need to return `$dv->geErrors()` directly or define `$this->errors = $dv->geErrors()` and return this ?

File: lizmap/modules/filter/classes/filterDatasource.class.php [hunk 2]
```diff
@@ -46,11 +46,11 @@ public function __construct($repository, $project, $layerId)
         jClasses::inc('filter~filterConfig');
         $dv = new filterConfig($repository, $project);
         if (!$dv->getStatus()) {
-            return $this->error($dv->getErrors());
+            return $dv->getErrors();
```
Comment (drillinP) [inline lizmap/modules/filter/classes/filterDatasource.class.php:49]: Same as previous comment

File: lizmap/modules/lizmap/classes/qgisVectorLayerDatasource.class.php [hunk 3]
```diff
@@ -33,6 +33,10 @@ class qgisVectorLayerDatasource
         'sql' => ' sql=(.*)$',
     );
 
+    protected $provider;
+
+    protected $datasource;
```
Comment (drillinP) [inline lizmap/modules/lizmap/classes/qgisVectorLayerDatasource.class.php:38]: **PhpStan** Say that ``` Access to an undefined property qgisVectorLayerDatasource::$provider. ``` and ```
 Access to an undefined property qgisVectorLayerDatasource::$datasource.``` [for this two line](https://github.com/3liz/lizmap-web-client/pull/2041/commits/76e291380881b866d6e079da0a8fbcbf0160c350#diff-520ea96b02586d47aa1d3ab69edf6f419050a52814f16d8f6e8e9a4f00d53bd7L51-L52)

File: lizmap/modules/lizmap/controllers/geobookmark.classic.php [hunk 4]
```diff
@@ -11,6 +11,8 @@
  */
 class geobookmarkCtrl extends jController
 {
+    protected $whiteParams;
```
Comment (drillinP) [inline lizmap/modules/lizmap/controllers/geobookmark.classic.php:14]: Here **PhpStan** fllagged the error `Access to an undefined property geobookmarkCtrl::$whiteParams.` so i think we need to define `protected $$whiteParams`

File: lizmap/modules/action/classes/actionDatasource.class.php (hunk 1 above)
Comment (rldhont) [reply lizmap/modules/action/classes/actionDatasource.class.php:44]: @mdouchin can you confirm ?

File: lizmap/modules/filter/classes/filterDatasource.class.php (hunk 2 above)
Comment (rldhont) [reply lizmap/modules/filter/classes/filterDatasource.class.php:49]: @mdouchin can you confirm ?

**Review by rldhont — CHANGES_REQUESTED** [review]
Can you run php-cs-fixer to fix the test.

**Comment by rldhont** [comment]
@drillinP can you also rebase the branch ?

**Comment by Gustry** [comment]
You should include all the content of the action, CF https://github.com/3liz/lizmap-web-client/pull/2029/files with the cache mechanism.

**Review by rldhont — APPROVED** [review]

**Review by Gustry — APPROVED** [review]
Just LGTM because it's green for phpstan !
We can polish the action later.

I will let PHP devs the real review.

File: lizmap/modules/action/classes/actionDatasource.class.php (hunk 1 above)
Comment (mdouchin) [reply lizmap/modules/action/classes/actionDatasource.class.php:44]: I confirm the proposed fix is OK. By the way, it seems the whole class actionDatasource is never used in Lizmap ! I will review this module code for next point release !

File: lizmap/modules/filter/classes/filterDatasource.class.php (hunk 2 above)
Comment (mdouchin) [reply lizmap/modules/filter/classes/filterDatasource.class.php:49]: I confirm the fix is ok

**Review by mdouchin — APPROVED** [review]

**Pull request merged**